libraries/resource_providers/api300/c7000/scope_provider.rb

```diff
@@ -25,7 +25,12 @@ def change_resource_assignments
           @item.retrieve!
           values_to_add = parse_resource_values
           values_to_remove = parse_resource_values(:remove)
-          @item.change_resource_assignments(add_resources: values_to_add, remove_resources: values_to_remove)
+          return Chef::Log.info("#{@resource_name} '#{@name}' is up to date") if values_to_add.empty? && values_to_remove.empty?
+          Chef::Log.debug "Resources to be added: #{JSON.pretty_generate(values_to_add)}" unless values_to_add.empty?
+          Chef::Log.debug "Resources to be removed: #{JSON.pretty_generate(values_to_remove)}" unless values_to_remove.empty?
+          @context.converge_by "Performing resource assignments on #{@item['name']}" do
```

Comment:
This will probably work, but the behavior here is a little strange. The text in the converge_by param isn't output until after it's done with the block, so this will actually say "Performing resource assignments" after it's done. I typically set the text in the converge_by param to the text I want to see after it's done successfully (ie "Assigned scope to resources..."). I also add an info log message just before it converges saying what is about to be done (ie, performing, creating, updating, etc). You added an info message saying it's up to date if the values_to_add/remove are empty, but nothing when they're not (until afterwards). Can I suggest something like...

```ruby
return Chef::Log.info("#{@resource_name} '#{@name}' is up to date") if values_to_add.empty? && values_to_remove.empty?
diff = ''
diff << "\n  Adding: #{JSON.pretty_generate(values_to_add)}" unless values_to_add.empty?
diff << "\n  Removing: #{JSON.pretty_generate(values_to_remove)}" unless values_to_remove.empty?
Chef::Log.info "Performing resource assignments on #{@resource_name} '#{@item['name']}':#{diff}"
@context.converge_by "Perform resource assignments on #{@resource_name} '#{@item['name']}'" do
  @item.change_resource_assignments(add_resources: values_to_add, remove_resources: values_to_remove)
end
```